### PR TITLE
Add `--censor` switch for disabling source code display

### PR DESCRIPTION
**Pull request by Gabriel439**
Related to #1294

This currently only affects parse errors, but can later be extended
to type errors, too

### CODE REVIEW

File: dhall/src/Dhall/Util.hs [hunk 1]
```diff
@@ -81,3 +90,35 @@ insert expression =
...
+    let result = parser name inText
+
+    let censoredResult =
+            if censor then first Dhall.Parser.censor result else result
+
+    Dhall.Core.throws censoredResult
+
+-- | Convenient utility for retrieving an expression
+getExpression
+    :: Bool
```
Comment (sjakobi) [inline dhall/src/Dhall/Util.hs]: A dedicated `CensorMode` or similar would probably be more readable.

File: dhall/src/Dhall/Util.hs [hunk 2]
```diff
@@ -81,3 +90,35 @@ insert expression =
...
+    -- ^ Censor the source code?
+    -> Maybe FilePath
```
Comment (sjakobi) [inline dhall/src/Dhall/Util.hs]: `Nothing` arguments also smell a bit like boolean blindness to me. IMHO the straightforward fix would be to have both `getExpressionFromFile` and `getExpressionFromStdin`, but it's not important to me.

**Comment by sjakobi** [comment]
Many thanks for the much nicer types! :)

**Comment by Gabriel439** [comment]
@sjakobi: You're welcome! :slightly_smiling_face:

**Pull request merged**